frostsnapp/lib/wallet.dart

```diff
@@ -716,3 +846,107 @@ Uri getBlockExplorer(BitcoinNetwork network) {
     return Uri.parse("https://mempool.space/signet/");
   }
 }
+
+class BannerStateProvider extends InheritedWidget {
+  final bool showBanner;
+
+  const BannerStateProvider({
+    required this.showBanner,
+    required Widget child,
+    Key? key,
+  }) : super(key: key, child: child);
+
+  static BannerStateProvider? of(BuildContext context) {
+    return context.dependOnInheritedWidgetOfExactType<BannerStateProvider>();
+  }
+
+  @override
+  bool updateShouldNotify(BannerStateProvider oldWidget) {
+    return showBanner != oldWidget.showBanner;
+  }
+}
```

Comment:
This `InheritedWidget` is just here so that we can get the initial state of whether we should or should not show the banner?

However, we set the `showBanner` value when we construct `WalletHomeWithConfetti` and then build the widgets by "wrapping" it in this `BannerStateProvider`.

I think it makes more sense just to not construct the `BackupWarningBanner` if we don't have `showBanner`, and pass the `showBanner` into `WalletHome` when we construct it.

Reply:
Okay I see why you wanted to use an `InheritedWidget`. It's because otherwise, we need to pass the `showBanner` variable through a few layers of widgets.

Reply:
I think we should merge `TxList` and `WalletHome` as a follow up and just pass `showBanner` via the constructors for now.

Reply:
Yeah very hacky sorry! I should've left a comment. Agree, passing via constructors does sound much better - worth trying in this PR?